gpii/node_modules/lifecycleManager/src/UserLogonStateChange.js

```diff
@@ -25,9 +25,24 @@ fluid.registerNamespace("gpii.lifecycleManager.userLogonHandling");
 gpii.lifecycleManager.userLogonHandling.handleResetGpiiKey = function (that, lifecycleManager) {
     var activeGpiiKeys = lifecycleManager.getActiveSessionGpiiKeys(); // find currently logged in user:
     var activeGpiiKey = activeGpiiKeys[0];
+    var defaultSettings = that.options.defaultSettings;
 
-    fluid.log("Reset: Logging out the user with GPII key " + activeGpiiKey);
-    that.logoutUser(activeGpiiKey);
+    var resetActions = [];
+    if (activeGpiiKey) {
+        fluid.log("Reset: logging out the user with GPII key " + activeGpiiKey);
+        resetActions.push(that.logoutUserWithoutResponse(activeGpiiKey));
+    }
+    if (defaultSettings) {
+        fluid.log("Reset: restoring the system to the default settings: ", defaultSettings);
+        resetActions.push(lifecycleManager.restoreSnapshot(defaultSettings));
+    }
+    var promise = fluid.promise.sequence(resetActions);
```

Comment:
I believe, because of the special behaviour of the "restore" key, that it is not necessary to queue a separate logout of "reset", but please double-check this and add a comment explaining why it is not necessary, and that the expected final condition of the system has no user logged in. If not, please add the "logout"